### PR TITLE
add LogMessage and move LogStabilityLevel

### PR DESCRIPTION
**Pull request by codeboten**
This change will make the func available for extensions as well.

### CODE REVIEW

File: service/internal/components/components.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// limitations under the License.
+
+package components // import "go.opentelemetry.io/collector/service/internal/components"
+
+import (
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+func LogStabilityLevel(logger *zap.Logger, sl component.StabilityLevel) {
```
Comment (mx-psi) [inline service/internal/components/components.go:22]: Could you add a docstring?

**Review by mx-psi — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
needs a rebase

File: service/internal/components/components_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+func TestLogStabilityLevle(t *testing.T) {
```
Comment (jpkrohling) [inline service/internal/components/components_test.go:28]: Can you also fix the typo from the original test?

```suggestion
func TestLogStabilityLevel(t *testing.T) {
```

File: component/component.go [hunk 3]
```diff
@@ -138,6 +138,24 @@ func (sl StabilityLevel) String() string {
...
+	case StabilityLevelDeprecated:
+		return "This component has been deprecated and will be removed in future releases."
+	case StabilityLevelInDevelopment:
+		return "This component is in development and may change in the future."
+	case StabilityLevelAlpha:
+		return "This component is alpha and may change in the future."
+	case StabilityLevelBeta:
+		return "This component is beta and may change in the future."
+	case StabilityLevelStable:
+		return "This component is stable."
```
Comment (bogdandrutu) [inline component/component.go:154]: I would rephrase things as "stable component", "unmaintained component and actively looking for contributors. This component will become deprecated after 6 months of remaining unmaintained."

Comment (TylerHelmuth) [reply component/component.go:154]: We could continue that syntax for the other cases:

- "Stable component."
- "Deprecated component. Will be removed in future releases."
- "Unmaintained component. Actively looking for contributors. This component will become deprecated after 6 months of remaining unmaintained."
- "In development component.  May change in the future."
- "Alpha component.  May change in the future."
- "Beta component.  May change in the future."

File: service/internal/components/components.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+package components // import "go.opentelemetry.io/collector/service/internal/components"
+
+import (
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// LogStabilityLevel logs the stability level of a component. The log level is set to info for
+// log levels: undefined, unmaintained, deprecated and in development. The log level is set to debug
```
[the remark this answers is not in the document]
Comment (codeboten) [reply service/internal/components/components.go]: Fixed

**Review by TylerHelmuth — APPROVED** [review]
LGTM
